src/widget/wcuemenupopup.cpp

```diff
@@ -135,6 +233,134 @@ void WCueMenuPopup::slotDeleteCue() {
     hide();
 }
 
+void WCueMenuPopup::slotStandardCue() {
+    VERIFY_OR_DEBUG_ASSERT(m_pCue != nullptr) {
+        return;
+    }
+    VERIFY_OR_DEBUG_ASSERT(m_pTrack != nullptr) {
+        return;
+    }
+    if (m_pCue->getType() != mixxx::CueType::HotCue) {
+        m_pCue->setType(mixxx::CueType::HotCue);
+    }
+    slotUpdate();
+}
+
+void WCueMenuPopup::slotSavedLoopCue() {
+    VERIFY_OR_DEBUG_ASSERT(m_pCue != nullptr) {
+        return;
+    }
+    VERIFY_OR_DEBUG_ASSERT(m_pTrack != nullptr) {
+        return;
+    }
+    VERIFY_OR_DEBUG_ASSERT(m_pBeatLoopSize.valid()) {
+        return;
+    }
+    if (m_pCue->getType() != mixxx::CueType::Loop) {
+        m_pCue->setType(mixxx::CueType::Loop);
+        auto cueStartEnd = m_pCue->getStartAndEndPosition();
+        if (!cueStartEnd.endPosition.isValid() ||
+                cueStartEnd.endPosition <= cueStartEnd.startPosition) {
+            double beatloopSize = m_pBeatLoopSize.get();
+            const mixxx::BeatsPointer pBeats = m_pTrack->getBeats();
+            if (beatloopSize <= 0 || !pBeats) {
+                return;
+            }
+            m_pCue->setEndPosition(pBeats->findNBeatsFromPosition(
+                    cueStartEnd.startPosition, beatloopSize));
+        }
+    }
+    slotUpdate();
+}
+
+void WCueMenuPopup::slotAdjustSavedLoopCue() {
+    VERIFY_OR_DEBUG_ASSERT(m_pCue != nullptr) {
+        return;
+    }
+    VERIFY_OR_DEBUG_ASSERT(m_pTrack != nullptr) {
+        return;
+    }
+    const mixxx::BeatsPointer pBeats = m_pTrack->getBeats();
+    auto position = mixxx::audio::FramePos::fromEngineSamplePos(
+            m_pPlayPos.get() * m_pTrackSample.get());
+    m_pCue->setType(mixxx::CueType::Loop);
+    if (!m_pQuantizeEnabled.toBool() || !pBeats) {
+        m_pCue->setEndPosition(position);
+    } else {
+        mixxx::audio::FramePos nextBeatPosition, prevBeatPosition;
+        pBeats->findPrevNextBeats(position, &prevBeatPosition, &nextBeatPosition, false);
+        position = (nextBeatPosition - position > position - prevBeatPosition)
+                ? prevBeatPosition
+                : nextBeatPosition;
+        if (position <= m_pCue->getPosition()) {
+            return;
+        }
+        m_pCue->setEndPosition(position);
+    }
+    slotUpdate();
+}
+
+void WCueMenuPopup::slotSavedJumpCue() {
+    VERIFY_OR_DEBUG_ASSERT(m_pCue != nullptr) {
+        return;
+    }
+    VERIFY_OR_DEBUG_ASSERT(m_pTrack != nullptr) {
+        return;
+    }
+    VERIFY_OR_DEBUG_ASSERT(m_pBeatJumpSize.valid()) {
+        return;
+    }
+    if (m_pCue->getType() != mixxx::CueType::Jump) {
+        auto cueStartEnd = m_pCue->getStartAndEndPosition();
+        if (!cueStartEnd.endPosition.isValid()) {
+            const mixxx::BeatsPointer pBeats = m_pTrack->getBeats();
+            const auto position = mixxx::audio::FramePos::fromEngineSamplePos(
+                    m_pPlayPos.get() * m_pTrackSample.get());
+            if (position == m_pCue->getPosition()) {
```

Comment:
For some reason this doesn't work for me:
when clicking Jump while at the cue position a zero-sized jump is created. When activating thus jump cue I run into an inifinite tiny loop.

Reply:
in the display start and end seem to be idetical.

Maybe a rounding issue with `fromEngineSamplePos()`?

Reply:
This should be fixed now